### PR TITLE
mdadm: migrate cron.weekly to systemd.timer

### PR DESCRIPTION
**Pull request by aniruddha2000**
# Replace cron.weekly with systemd.timer

Add a mdadm.timer unit and associate it with the mdadm package and remove the cron.weekly.

## Testing done

### CODE REVIEW

**Comment by aniruddha2000** [comment]
@pothos can you give me a little code review that what I am missing. Because when I have built an image the `systemctl status *timer` is not giving me the mdadm.timer

File: sys-fs/mdadm/files/mdadm.service [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[Unit]
+Description=Initiates a check run of an MD array's redundancy information.
+
+[Service]
+ExecStart=/usr/sbin/checkarray
```
Comment (pothos) [inline sys-fs/mdadm/files/mdadm.service:5]:
```suggestion
ExecStart=/usr/sbin/checkarray --cron --all --idle --quiet
```
I think we should keep the same flags

File: sys-fs/mdadm/mdadm-4.1.ebuild [hunk 2]
```diff
@@ -78,8 +78,7 @@ src_install() {
 	insinto /etc/default
 	newins "${FILESDIR}"/etc-default-mdadm mdadm
 
-	exeinto /etc/cron.weekly
-	newexe "${FILESDIR}"/mdadm.weekly mdadm
+	systemd_dounit "${FILESDIR}"/mdadm.service
```
Comment (pothos) [inline sys-fs/mdadm/mdadm-4.1.ebuild:81]:
```suggestion
	systemd_dounit "${FILESDIR}"/mdadm.service
	systemd_dounit "${FILESDIR}"/mdadm.timer
```
Both need to be added, I'll need to find how to enable the timer.

Comment (pothos) [reply sys-fs/mdadm/mdadm-4.1.ebuild:81]: I think `systemd_enable_service timers.target mdadm.timer`

File: sys-fs/mdadm/files/mdadm.service [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[Unit]
+Description=Initiates a check run of an MD array's redundancy information.
+
+[Service]
+ExecStart=/usr/sbin/checkarray --cron --all --idle --quiet
```
Comment (pothos) [inline sys-fs/mdadm/files/mdadm.service:5]:
```suggestion
Type=oneshot
ExecStart=/usr/sbin/checkarray --cron --all --idle --quiet
```
Making it a oneshot service unit also makes sense (cf. , e.g.,`fstrim.service` as shipped in Fedora/Debian)

**Comment by pothos** [comment]
I started a test build for this (build id is 3483)

**Review by pothos — APPROVED** [review]

**Comment by pothos** [comment]
Thanks, the timer works

**Pull request merged**